src/Compilers/Core/Portable/InternalUtilities/Debug.cs

```diff
@@ -13,15 +14,21 @@ internal static class RoslynDebug
     {
         /// <inheritdoc cref="Debug.Assert(bool)"/>
         [Conditional("DEBUG")]
-        public static void Assert([DoesNotReturnIf(false)]bool b) => Debug.Assert(b);
+        public static void Assert([DoesNotReturnIf(false)] bool b) => Debug.Assert(b);
 
         /// <inheritdoc cref="Debug.Assert(bool, string)"/>
         [Conditional("DEBUG")]
-        public static void Assert([DoesNotReturnIf(false)]bool b, string message)
+        public static void Assert([DoesNotReturnIf(false)] bool b, string message)
             => Debug.Assert(b, message);
 
         [Conditional("DEBUG")]
-        public static void AssertNotNull<T>([NotNull]T value) where T : class?
-            => Debug.Assert(value is object, "Unexpected null reference");
+        public static void AssertNotNull<T>([NotNull] T value) where T : class?
+        {
+            Debug.Assert(value is object, "Unexpected null reference");
+            if (value is null)
+            {
+                throw new InvalidOperationException("This program location is unreachable.");
+            }
```

Comment:
Is the `throw` needed since the assert above will have failed?

Reply:
The assert above is not necessarily annotated, in some of our projects.

Reply:
Could you implement in terms `RoslynDebug.Assert` instead of `Debug.Assert` to solve this?

Reply:
Ah, indeed :-) I'll try using `RoslynDebug.Assert`